### ISSUE TITLE
Bug: Using a plug-in to add outfit space also grants extra weapons

### ISSUE THREAD

**Issue by Zitchas** [endless-sky/endless-sky#5649]
**Describe the bug**
When I am using a plug-in that adds outfit space to the standard existing Albatross, when I purchase an Albatross (which is listed in the shipyard as having 6 inhibitor canons, 2 turrets, 1 PDT) but then look at the player info screen, and it shows all 7 canons slots filled with inhibitor canons, and the turret slots filled with an extra thrasher turret and PDT. The outfit list still says exactly what the shipyard said it should have. Selling *all* weapons and taking off and firing results in me shooting, and I have the extra weapons in their spots, and they fire. Reloading the game causes the game to flag the extra outfits as "not existing in the outfit list" and removes them.

**To Reproduce**
1. Make a plug-in with the following code:
```
ship "Albatross"
    add attributes
        "outfit space" 34
```
or
```
ship "Albatross"
    attributes
        "outfit space" 34
```
1B. Run Endless Sky.
2. Go to a Remnant planet
3. Buy an Albatross
4. Before leaving shipyard, note that the shipyard lists 6 inhibitor canons, 2 thrasher turrets, and 1 PDT.
5. Leave shipyard
6. Check player info. Note that the Player Info outfit lists shows the correct numbers of outfits.
7. Note that the ship pictogram showing weapon locations has all seven canon slots filled with inhibitor canons. Likewise, 3 of the turret slots have thrasher turrets, and 2 of the turret slots have PDT.
8. Launch, and verify that all the weapons do actually work.
9. Land, and sell all the weapons.
10. take off, and note that the extra canon and two turrets are functional
11. Reload the game, note that the game gives errors related to those three outfits, and removes them

**Expected behavior**
The official "add attribute" code, used as described in the "plugin creation section" should not cause extra weapons to be added to the target ship.

**Screenshots**
![Uh-guns-turrets](https://user-images.githubusercontent.com/32169904/104824382-63acfd80-580e-11eb-94aa-645cf33064a4.PNG)

**System (please complete the following information):**
 - OS: Windows 10 Pro
 - Game Source: github compiled version on 2021-Jan-15
 - Version: 

**Link to save file**
[e.g. hastebin.com, direct upload, GitHub Gist]

**Additional context**
Possibly related problem: #4413

**Comment by vitalchip** [comment]
I think this come down to trying to overwrite a ship by using a system meant for creating variants. Insufficient information is being passed to fully overwrite the ship and what's missing is being populated from an earlier loaded instance.

The code attached is where the problem manifests, even though the hardpoint definitions get copied from the previous instance (because they're not being specified in the new instance) the 'equipped' list for the new ship instance is empty. So when it finds e.g. 2x Thrasher Turrets in the outfits list (also copied from previous ship instance) it compares to the equipped list for the new ship (empty) finds a difference and passes to armament to install, filling empty hardpoints.

The extra weapons will vanish after re-loading because next time round when the ship is loaded from player save file there will be a mis-match which will be corrected.

This seems more like a 'missing' feature than anything else i.e. there isn't a 'shortcut' for over writing ships.

You can 'get away' with missing out the outfits list and just changing the attributes e.g. add cargo space, shields etc but you will have to define the hardpoints (fully) so that the new ship instance has an equipped list to compare against.

<img width="586" alt="Screenshot 2021-01-17 at 00 58 01" src="https://user-images.githubusercontent.com/65342577/104844328-a8489f80-58c7-11eb-9559-702f3e167a56.png">

**Comment by tehhowch** [comment]
duplicate of #4413, fixed by #6216 as well.
